src/o365/spo/commands/page/page-clientsidewebpart-add.ts

```diff
@@ -0,0 +1,483 @@
+			throw new Error(`Invalid Section '${args.options.section}'`);
+		}
+
+		// Get the target section
+		const section: CanvasSection = clientSidePage.sections[actualSectionIndex];
+
+		let actualColumnIndex: number | null = args.options.column && args.options.column - 1;
+		// If the column arg is not specified, set to first column
+		if (actualColumnIndex == null) {
+			if (this.debug) {
+				cmd.log(`No column argument specified, The component will be added to default column`);
+			}
+			actualColumnIndex = 0;
+		}
+		// Make sure the column is in the range of the current section
+		if (actualColumnIndex >= section.columns.length) {
+			throw new Error(`Invalid Column '${args.options.column}'`);
+		}
+
+		// Get the target column
+		const column: CanvasColumn = section.columns[actualColumnIndex];
+
+		if (args.options.webPartProperties) {
+			if (this.debug) {
+				cmd.log('WebPart properties: ');
+				cmd.log(args.options.webPartProperties);
+				cmd.log('');
+			}
+
+			try {
+				let properties = JSON.parse(args.options.webPartProperties);
+				webPart.setProperties(properties);
+			} catch (error) {
+				throw new Error('WebPart properties cannot be set');
+			}
+		}
+
+		// Add the WebPart to to appropriate location
+		column.addControl(webPart);
+	}
+
+	private _saveClientSidePage(
+		clientSidePage: ClientSidePage,
+		cmd: CommandInstance,
+		args: CommandArgs,
+		accessToken: string,
+		requestDigest: string
+	): request.RequestPromise {
+		let pageName: string = args.options.pageName;
+		if (!pageName.endsWith('.aspx')) {
+			pageName += '.aspx';
+		}
+		const serverRelativeSiteUrl: string = `${args.options.webUrl.substr(
+			args.options.webUrl.indexOf('/', 8)
+		)}/sitepages/${pageName}`;
+
+		const updatedContent = clientSidePage.toHtml();
+
+		if (this.debug) {
+			cmd.log('Updated canvas content: ');
+			cmd.log(updatedContent);
+			cmd.log('');
+		}
+
+		const requestOptions: any = {
+			url: `${args.options
+				.webUrl}/_api/web/getfilebyserverrelativeurl('${serverRelativeSiteUrl}')/ListItemAllFields`,
+			headers: Utils.getRequestHeaders({
+				authorization: `Bearer ${accessToken}`,
+				'X-RequestDigest': requestDigest,
+				'content-type': 'application/json;odata=nometadata',
+				'X-HTTP-Method': 'MERGE',
+				'IF-MATCH': '*',
+				accept: 'application/json;odata=nometadata'
+			}),
+			body: {
+				CanvasContent1: updatedContent
+			},
+			json: true
+		};
+
+		if (this.debug) {
+			cmd.log('Executing web request...');
+			cmd.log(requestOptions);
+			cmd.log('');
+		}
+
+		return request.post(requestOptions);
+	}
+
+	public options(): CommandOption[] {
+		const options: CommandOption[] = [
+			{
+				option: '-n, --pageName <pageName>',
+				description: 'Name of the page where the control is located'
+			},
+			{
+				option: '-u, --webUrl <webUrl>',
+				description: 'URL of the site where the page to retrieve is located'
+			},
+			{
+				option: '--standardWebPart <standardWebPart>',
```

Comment:
`standardWebPart` is optional, so we can use `[standardWebPart]` instead of `<standardWebPart>`. Here it's just notation, but it's a common convention in command line tools that we follow in the CLI as well